---
layout: frontpage
title: Ben Li
description: Ben Li is currently a Ph.D. candidate in the Department of Biostatistics and Bioinformatics at Emory University 
keywords: Ben Li, Emory,  Bioinformatics, Bayesian Statistics, Machine Learning, NGS

---

<div class="navbar">
  <div class="navbar-inner">
      <ul class="nav">
          <li><a href="https://scholar.google.com/citations?user=nDSGBakAAAAJ&hl=en">Google Scholar</a></li>
          <li><a href="https://github.com/benliemory">Github</a></li>
      </ul>
  </div>
</div>

<p>
I am currently a Ph.D. candidate in the Department of Biostatistics and Bioinformatics at Emory University and will complete my degree in May 2017. My dissertation is under the supervision of Professors Zhaohui Steve Qin and Hao Wu. I respectfully submit this letter of application, for I believe my research experiences and teaching commitment make me well qualified to meet the requirements of your program.
My research mainly focuses on the development of statistical and computational methods for high- throughput genetics and genomics data. During my Ph.D. studies, I have extensive research experiences on developing novel statistical and computational methods for next generation sequencing (NGS) data. I have published first author (co-first author) methodological papers on hierarchical models incorporating historical data to detect differentially expressed gene on microarray data (Bioinformatics, 2016), detecting differentially methylated regions on 450K methylation array and bisulfite sequencing data (Statistics in Biosciences, 2016), novel supervised learning approach to predict TFâ€“DNA interaction using bisulfite sequencing methylation data (Nucleic acids research, 2015). I am currently working on developing new statistical methods for detecting differentially expressed genes without replicates for both microarray and RNA-seq data.
Moreover, I have worked together with collaborators of different backgrounds and created computational tools to help collaborators analyze different types of high-throughput -omics data more efficiently. For example, I developed binstrain for Chlamydia trachomatis Strains in Single and Mixed Infection Clinical Samples (PloS one, 2014) and for detecting Staphylococcus aureus subtypes from metagenome shotgun sequence data (PeerJ, 2016). My other collaborative projects include analyses of bisulfite sequencing data (Nucleic acids research, 2016), Hi-C data (Statistics in Biosciences, 2016), single cell RNA-seq data, and ChIP-seq data. At Emory, I also provided statistical consulting services to assist principal investigators from Winship Cancer Institute and Yerkes National Primate Research Center to design experiments and analyze high-throughput -omics data.
</p>



<!--

<table class="wide">
<tr>
  <td class="left">
    <a href="pages/publpics/iplotCorr.html">
        <img src="assets/publpics/iplotCorr.png" alt="R/qtlcharts example" title="R/qtlcharts example"/>
    </a>
  </td>
  <td class="right">
    <a href="pages/publpics/tian2016_fig4.html">
        <img src="assets/publpics/tian2016_fig4.png" alt="Tian et
        al. (2016) Fig 4" title="Tian et al. (2016) Fig 4"/>
    </a>
  </td>
</tr>
<tr>
  <td class="left">
    <a href="pages/publpics/samplemixups_fig7.html">
        <img src="assets/publpics/samplemixups_fig7.png" alt="Broman et al. (2013) Fig 7" title="Broman et al. (2013) Fig 7"/>
    </a>
  </td>
  <td class="right">
    <a href="pages/publpics/isletc6_fig4.html">
        <img src="assets/publpics/isletc6_fig4.png" alt="Tian et al. (2015) Fig 4" title="Tian et al. (2015) Fig 4"/>
    </a>
  </td>
</tr>
</table>

<div class="navbar">
  <div class="navbar-inner">
      <ul class="nav">
          <li><a href="morefigs.html">see more figures</a></li>
      </ul>
  </div>
</div>

-->
