---
layout: page
title: Software
description: Ben Li's software
---

<div class="navbar">
    <div class="navbar-inner">
        <ul class="nav">
            <li><a href="https://github.com/benliemory">Github</a></li>
            <li><a href="https://github.com/benliemory/IPBT">IPBT</a></li>
            <li><a href="https://github.com/benliemory/adaptiveHM">adaptiveHM</a></li>
            <li><a href="https://github.com/benliemory/Methylphet">Methylphet</a></li>
            <li><a href="https://github.com/benliemory/BinStrain">BinStrain</a></li>
        </ul>
    </div>
</div>

---

#### <a name="IPBT"></a>[IPBT](https://github.com/benliemory/IPBT)

**I**nformative **P**rior **B**ayesian **T**est

R package is provided in the IPBT_R_package folder. All the IPBT priors are built-in features in IPBT package.

All the IPBT priors are also summarized in the Informative_Priors folder in excel sheets. Although only variance is used for IPBT priors, we also provide the historical means for possible use.

---

#### <a name="adaptiveHM"></a>[adaptiveHM](https://github.com/benliemory/adaptiveHM)

adaptiveHM presents a new framework to enhance the Bayesian hierarchical model. The historical Rank-based adaptive hierarchical model enables borrowing information across different platforms which could be extremely useful with emergence of new technologies and accumulation of data from different platforms in the Big Data era.

---

#### <a name="Methylphet"></a>[Methylphet](https://github.com/benliemory/Methylphet)

Methylphet adopts a two-step method to predict transcription factor-DNA interaction using DNA methylation profiles from whole-genome bisulfite sequencing data by exploiting the connection between DNA methylation level and transcription factor binding. In the first step, beta-binomial models are devised to characterize DNA methylation data around TF binding sites and the background to estimate methylation scores. Along with other static genomic features, a random forest framework is adopted in the second step to predict transcription factor-DNA interaction. When all methylation profile are taken together and combined with features at the sequence level, Methylphet can accurately predict TF binding and performs favorably when compared against competing methods.  

---

#### <a name="BinStrain"></a>[BinStrain](https://github.com/benliemory/BinStrain)

Various examples of [D3](http://d3js.org)-based interactive graphs.

---


